### PR TITLE
Fix item validity check for transfer

### PR DESCRIPTION
**Pull request by SNBeast**
Also de-dupes code in PK1->PK2 and adds item transfer for PK3->PK2

### CODE REVIEW

File: source/sav/Sav.cpp [hunk 1]
```diff
@@ -333,33 +334,55 @@ namespace pksm
         {
             return BadTransferReason::MOVE;
         }
-        else if (availableSpecies().count(pk.species()) == 0)
+        if (availableSpecies().count(pk.species()) == 0)
```
Comment (piepie62) [inline source/sav/Sav.cpp:337]: Why did you remove these `else`s?

Comment (SNBeast) [reply source/sav/Sav.cpp:337]: They didn't have to be removed per se, but removing them means that if the function has to be maintained more, then potential false negatives from skipped evaluations can be avoided. Consider the abilities check, for example.

Comment (piepie62) [reply source/sav/Sav.cpp:337]: Ah, fair enough, makes sense
